### PR TITLE
add homepae & carousel & update nav & fix small bugs

### PR DESCRIPTION
**Pull request by zhangtianli2006**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,30 +1,30 @@
 <template>
   <div id="app">
-    <NavBar/>
+    <NavBar />
```
Comment (szdytom) [inline src/App.vue:3]: Due to W3C, **dont't delete this space**

File: src/App.vue [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
     <div id="main">
       <div id="content">
         <router-view></router-view>
       </div>
     </div>
-    <Footer/>
+    <Footer />
```
Comment (szdytom) [inline src/App.vue:9]: Same Problem

File: src/App.vue [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 
 <script>
-import NavBar from './components/page/navbar.vue';
-import Footer from './components/page/footer.vue';
-import UserLogin from './components/user/login.vue';
-import UserRegister from './components/user/register.vue';
+import NavBar from "./components/page/navbar.vue";
+import Footer from "./components/page/footer.vue";
+import UserLogin from "./components/user/login.vue";
+import UserRegister from "./components/user/register.vue";
```
Comment (szdytom) [inline src/App.vue:19]: We use `''`

File: src/components/404.vue [hunk 4]
```diff
@@ -24,4 +24,8 @@ export default {
   font-size: 2em;
   font-weight: 100;
 }
+
+.box {
+  text-align: center;
+}
```
Comment (szdytom) [inline src/components/404.vue:30]: May be you can add this to a global css file

File: src/components/homepage/carousel.vue [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+.el-carousel__item:nth-child(2n) {
+  background-color: #99a9bf;
+}
+
+.el-carousel__item:nth-child(2n + 1) {
+  background-color: #d3dce6;
+}
+</style>
```
Comment (szdytom) [inline src/components/homepage/carousel.vue:25]: You didn't add any thing here yet, did you?

File: src/components/lib/AjaxTable.vue [hunk 6]
```diff
@@ -1,30 +1,33 @@
...
-      >
-    </el-pagination>
+    ></el-pagination>
   </div>
 </template>
 
 <script>
 export default {
-  name: 'AjaxTable',
+  name: "AjaxTable",
```
Comment (szdytom) [inline src/components/lib/AjaxTable.vue:24]: `''` here please

File: src/components/lib/AjaxTable.vue [hunk 7]
```diff
@@ -34,21 +37,21 @@ export default {
...
+        })
+        .then(res => {
+          console.log(res);
+          this.tableData = res.data.res.map(this.process);
+          this.loading = false;
+        })
+        .catch(err => {
+          this.$message.error("[Ajax Table] Request Failed");
+          console.log(err);
+        });
```
Comment (szdytom) [inline src/components/lib/AjaxTable.vue:54]: I think align the `.` is better(May be you should change a formatter
```js
this.xxx()
    .xxxxx()
    .xxxxxx()
```

File: src/components/page/navbar.vue [hunk 8]
```diff
@@ -1,28 +1,29 @@
...
               <MyUserName />
             </el-menu-item>
-            <el-menu-item index="/user/logout">Logout</el-menu-item>
+            <el-menu-item index='/user/logout'>Logout</el-menu-item>
           </div>
           <div v-else>
-            <el-menu-item @click="$store.state.user.showlogin = true;">Login</el-menu-item>
-            <el-menu-item @click="$store.state.user.showregister = true;">Register</el-menu-item>
+            <el-menu-item @click='$store.state.user.showlogin = true;'>Login</el-menu-item>
+            <el-menu-item @click='$store.state.user.showregister = true;'>Register</el-menu-item>
```
Comment (szdytom) [inline src/components/page/navbar.vue:26]: We use `""` in HTML and `''` in js

File: src/components/page/navbar.vue [hunk 9]
```diff
@@ -46,8 +47,9 @@ export default {
 };
 </script>
 
-<style lang="css" scoped>
+<style lang='css' scoped>
```
Comment (szdytom) [inline src/components/page/navbar.vue:50]: `""` here

**Comment by zhangtianli2006** [comment]
Damn you, formater!

File: src/components/404.vue (hunk 4 above)
Comment (zhangtianli2006) [reply src/components/404.vue:30]: Maybe you didn't notice that there is a ' . ' before 'box'
this don't need a fix

File: src/components/homepage/carousel.vue (hunk 5 above)
Comment (zhangtianli2006) [reply src/components/homepage/carousel.vue:25]: Yeah, maybe later you should add a picture function in the back-end
leave it here, maybe change it after

File: src/components/lib/AjaxTable.vue (hunk 6 above)
Comment (zhangtianli2006) [reply src/components/lib/AjaxTable.vue:24]: I bat FISHER_ wrote this

File: src/components/page/navbar.vue (hunk 8 above)
Comment (zhangtianli2006) [reply src/components/page/navbar.vue:26]: hmm...

**Comment by zhangtianli2006** [comment]
Can't we just use only " or only ' ??

File: src/components/homepage/carousel.vue (hunk 5 above)
Comment (szdytom) [reply src/components/homepage/carousel.vue:25]: ok

File: src/App.vue [hunk 10]
```diff
@@ -1,8 +1,8 @@
 <template>
-  <div id="app">
+  <div id='app'>
     <NavBar/>
-    <div id="main">
-      <div id="content">
+    <div id='main'>
+      <div id='content'>
```
Comment (szdytom) [inline src/App.vue:5]: I said that **we use `""`in HTML and `''` in js**

File: src/components/page/navbar.vue [hunk 11]
```diff
@@ -12,11 +12,19 @@
...
+                <MyUserName/>
+              </el-avatar>
+              <el-avatar v-else>
+                User
+              </el-avatar>
+          </template>
           <div v-if="this.$store.state.user.authenticated">
             <el-menu-item>
-              <MyUserName />
+              <MyUserName/>
```
Comment (szdytom) [inline src/components/page/navbar.vue:27]: lost a space

**Comment by szdytom** [comment]
Why we use `""` in HTML and `''` in js?
1. use `""` in HTML is defined in W3C & Very commen
1. when write inline script in HTML, different symbol in important
example:
```html
<p onclick="alert('Click!');">xxx</p>
```

but if we use same symbol:

```html
<p onclick="alert(\"Click!\");">xxx</p>
```
not beautiful...

**Pull request merged**